### PR TITLE
TW-192 Load more contact for Contact tab

### PR DESCRIPTION
**Pull request by hoangdat**
https://github.com/linagora/twake-on-matrix/assets/43041967/7c24ca85-1604-455b-a711-924444b63907

https://github.com/linagora/twake-on-matrix/assets/43041967/26331ebd-dae5-485a-bbd0-a3c5179e1875

### CODE REVIEW

File: lib/domain/model/extensions/contact/contact_extension.dart [hunk 1]
```diff
@@ -8,8 +8,17 @@ extension ContactExtension on Contact {
       displayName: displayName,
       matrixId: matrixId,
       status: status,
-    ),);
+    ),).toSet() ?? {};
```
Comment (nqhhdev) [inline lib/domain/model/extensions/contact/contact_extension.dart:11]:
```suggestion
    )).toSet() ?? {};
```

File: lib/domain/model/extensions/contact/contact_extension.dart [hunk 2]
```diff
@@ -8,8 +8,17 @@ extension ContactExtension on Contact {
...
 
-    return listContacts?.toSet() ?? {};
+    if (emails == null || emails!.isEmpty) {
+      listContacts.add(PresentationContact(
+        email: null,
```
Comment (nqhhdev) [inline lib/domain/model/extensions/contact/contact_extension.dart:15]: What will happen?

File: lib/domain/usecase/load_more_internal_contacts.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import 'package:fluffychat/domain/app_state/contact/get_contacts_success.dart';
+import 'package:fluffychat/domain/app_state/contact/get_network_contact_failed.dart';
+import 'package:fluffychat/domain/app_state/contact/get_network_contact_success.dart';
+import 'package:fluffychat/domain/model/contact/contact_query.dart';
+import 'package:fluffychat/domain/repository/contact_repository.dart';
+
+class LoadMoreInternalContacts {
+  final _contactRepository = getIt.get<ContactRepository>();
+
+  Stream<Either<Failure, GetContactsSuccess>> execute({int limit = 30, int? offset = 0}) async* {
```
Comment (nqhhdev) [inline lib/domain/usecase/load_more_internal_contacts.dart:13]: `const int limitInternalContacts = 30;`

File: lib/domain/usecase/load_more_internal_contacts.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    try {
+      final contacts = await _contactRepository
+        .loadMoreContact(
+          query: ContactQuery(keyword: ''),
```
Comment (nqhhdev) [inline lib/domain/usecase/load_more_internal_contacts.dart:17]: - Can be optional? WDYT?

File: lib/pages/new_group/widget/contacts_selection_list.dart [hunk 5]
```diff
@@ -51,55 +50,74 @@ class _ContactsSelectionListState extends State<ContactsSelectionList> {
...
+              builder: (context, isSearchMode, child) {
+                if (isSearchMode) {
+                  return const SizedBox.shrink();
+                }
+                return ValueListenableBuilder(
+                  valueListenable: fetchContactsController.haveMoreCountactsNotifier,
+                  builder: (context, value, child) {
+                    if (value) {
+                      return const Padding(
+                        key: ValueKey('new_group_chat'),
```
Comment (nqhhdev) [inline lib/pages/new_group/widget/contacts_selection_list.dart:107]: make const

File: lib/pages/new_private_chat/widget/expansion_list.dart [hunk 6]
```diff
@@ -111,24 +106,61 @@ class _ExpansionList extends State<ExpansionList> {
...
+                itemCount: contactsListSorted.length,
+                itemBuilder: (context, index) {
+                  final contact = contactsListSorted[index];
+                  return InkWell(
+                    onTap: () {
+                      widget.newPrivateChatController.goToChatScreen(contact: contact);
                     },
-                  ); 
+                    borderRadius: BorderRadius.circular(16.0),
+                    child: ExpansionContactListTile(contact: contact,),
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/expansion_list.dart:126]:
```suggestion
                    child: ExpansionContactListTile(contact: contact),
```

File: lib/pages/new_private_chat/widget/expansion_list.dart [hunk 7]
```diff
@@ -111,24 +106,61 @@ class _ExpansionList extends State<ExpansionList> {
...
+            }),
+          ),
+          ValueListenableBuilder(
+            valueListenable: searchContactsController.isSearchModeNotifier,
+            builder: (context, isSearchMode, child) {
+              if (isSearchMode) {
+                return const SizedBox.shrink();
+              }
+              return ValueListenableBuilder(
+                valueListenable: widget.newPrivateChatController.isShowContactsNotifier ,
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/expansion_list.dart:139]:
```suggestion
                valueListenable: widget.newPrivateChatController.isShowContactsNotifier,
```

File: lib/pages/new_private_chat/widget/expansion_list.dart [hunk 8]
```diff
@@ -111,24 +106,61 @@ class _ExpansionList extends State<ExpansionList> {
...
+                builder: (context, isShow, child) {
+                  if (!isShow) {
+                    return const SizedBox.shrink();
+                  }
+                  return ValueListenableBuilder(
+                    valueListenable: fetchContactsController.haveMoreCountactsNotifier,
+                    builder: (context, value, child) {
+                      if (value) {
+                        return const Padding(
+                          key: ValueKey('new_private_chat'),
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/expansion_list.dart:149]: make const

File: lib/domain/model/extensions/contact/contact_extension.dart (hunk 2 above)
Comment (hoangdat) [reply lib/domain/model/extensions/contact/contact_extension.dart:15]: no problem, some people maybe dont have email in contact

File: lib/domain/usecase/load_more_internal_contacts.dart (hunk 4 above)
Comment (hoangdat) [reply lib/domain/usecase/load_more_internal_contacts.dart:17]: take it in later, we can reuse `loadmore` for search also here. But ignore it now

File: lib/pages/new_private_chat/widget/expansion_list.dart (hunk 8 above)
Comment (hoangdat) [reply lib/pages/new_private_chat/widget/expansion_list.dart:149]: what the key? really?

**Pull request merged**